cpu/load_balancer.py

```diff
@@ -0,0 +1,162 @@
+
+import multiprocessing
+import os
+import re
+import time
+import pandas as pd
+from avocado import Test
+from avocado.utils import process, cpu, distro
+from avocado.utils.software_manager.manager import SoftwareManager
+
+totalcpus = int(multiprocessing.cpu_count()) - 1
+file_path = "/tmp/mpstat.log"
+online_cpu_check = "/sys/devices/system/cpu/"
+cpu_payload_data = []
+if os.path.exists(file_path):
+    os.remove(file_path)
+
+
+class load_balancer(Test):
+    """
+    _test summary_
+    -> This experiment entails executing a workload and subsequently
+        analyzing how a load balancer distributes the workload among
+        the available CPU cores and validate the functionality
+        of load balancer.
+    """
+    def setUp(self):
+        if 'ppc' not in distro.detect().arch:
+            self.cancel("Processor is not powerpc")
+        sm = SoftwareManager()
+        self.detected_distro = distro.detect()
+        if not sm.check_installed("powerpc-utils") and \
+                not sm.install("powerpc-utils"):
+            self.cancel("powerpc-utils is needed for the test to be run")
+        smt_op = process.run("ppc64_cpu --smt", shell=True,
+                             ignore_status=True).stderr.decode("utf-8")
+        if "is not SMT capable" in smt_op:
+            self.cancel("Machine is not SMT capable, skipping the test")
+        self.no_threads = self.params.get("no_threads", default=24)
+        self.cpu_cycles = self.params.get("cpu_cycles", default=10000000)
+        self.capacity = self.params.get("capacity", default=100)
+        distro_name = self.detected_distro.name
+        distro_ver = self.detected_distro.version
+        distro_rel = self.detected_distro.release
+        if distro_name == "rhel":
+            if (distro_ver == "7" or
+                    (distro_ver == "8" and distro_rel < "4")):
+                self.cancel("smtstate tool is supported only after RHEL8.4")
+        elif distro_name == "SuSE":
+            if (distro_ver == "12" or (distro_ver == "15" and distro_rel < 3)):
+                self.cancel("smtstate tool is supported only after SLES15 SP3")
+        else:
+            self.cancel("Test case is supported only on RHEL and SLES")
+        process.run("pip3 install pandas", shell=True)
+
+    def run_workload(self, no_threads, cpu_cycles, capacity):
+        """
+        Run the stress-ng workload
+        """
+        cmd = 'nohup stress-ng --cpu %s --cpu-ops %s -l %s &> \
+                /tmp/stress-ng.txt &' % (no_threads, cpu_cycles, capacity)
+        process.run(cmd, shell=True)
+        self.log.info("stress-ng workload started running successfully--!!")
+
+    def mpstat_analyzer(self, on_cpu_count, smt_mode, core):
+        """
+        This function basically capture the mpstat command
+        output for cpu utilization and validate the mpstat data.
+        """
+        cmd = 'mpstat -P ALL 1 1 | awk \'$3 != "all" && $4 > 0\' &> \
```

Comment:
make sure to install stress-ng and mpstat comands are installed, as some distro does not have this by default

Reply:
> make sure to install stress-ng and mpstat comands are installed, as some distro does not have this by default

Yes, I will add the install check for stress-ng and mpstat in setup.